espresso/core/java/androidx/test/espresso/matcher/ViewMatchers.java

```diff
@@ -1381,7 +1385,7 @@ protected void describeMoreTo(Description description) {
         description.appendText(" [").appendText(resourceName).appendText("]");
       }
       if (null != expectedText) {
-        description.appendText(" value: ").appendText(expectedText);
+        description.appendText(" value: ").appendText(highlightNbsp(expectedText));
```

Comment:
There are times when devs write tests and if the expectation mismatches, they use the failure message to update the test or the production code. In this case doing a replacement like this will cause headaches as copying the "actual" to the test will not make the test pass.

Since we're in a matcher we can highlight things better.
```suggestion
        description.appendText(" value: ").appendText(highlightNbsp(expectedText))
                   .appendText("; raw value: ").appendText(expectedText);
```

Reply:
May have put it in the wrong place, but it I think you get the point. This applies to expected and actual in all places.

Reply:
That's a good point.

However, placing the same string twice at a straight may cause some difficulties while reading. Besides, having these strings you still don't know what's the real problem is.
I added notation instead.